lib/fluent/plugin/out_bigquery.rb

```diff
@@ -2,374 +2,303 @@
+
+      # dataset_name
+      #   The name can be up to 1,024 characters long, and consist of A-Z, a-z, 0-9, and the underscore,
+      #   but it cannot start with a number or underscore, or have spaces.
+      config_param :dataset, :string
+
+      # table_id
+      #   In Table ID, enter a name for your new table. Naming rules are the same as for your dataset.
+      config_param :table, :string, default: nil
+      config_param :tables, :string, default: nil # TODO: use :array with value_type: :string
+
+      # template_suffix (only insert)
+      #   https://cloud.google.com/bigquery/streaming-data-into-bigquery#template_table_details
+      config_param :template_suffix, :string, default: nil
+
+      config_param :auto_create_table, :bool, default: false
+
+      # skip_invalid_rows (only insert)
+      #   Insert all valid rows of a request, even if invalid rows exist.
+      #   The default value is false, which causes the entire request to fail if any invalid rows exist.
+      config_param :skip_invalid_rows, :bool, default: false
+      # max_bad_records (only load)
+      #   The maximum number of bad records that BigQuery can ignore when running the job.
+      #   If the number of bad records exceeds this value, an invalid error is returned in the job result.
+      #   The default value is 0, which requires that all records are valid.
+      config_param :max_bad_records, :integer, default: 0
+      # ignore_unknown_values
+      #   Accept rows that contain values that do not match the schema. The unknown values are ignored.
+      #   Default is false, which treats unknown values as errors.
+      config_param :ignore_unknown_values, :bool, default: false
+
+      config_param :schema_path, :string, default: nil
+      config_param :fetch_schema, :bool, default: false
+      config_param :fetch_schema_table, :string, default: nil
+      config_param :schema_cache_expire, :time, default: 600
+      config_param :field_string,  :string, default: nil
+      config_param :field_integer, :string, default: nil
+      config_param :field_float,   :string, default: nil
+      config_param :field_boolean, :string, default: nil
+      config_param :field_timestamp, :string, default: nil
+      ### TODO: record field stream inserts doesn't works well?
+      ###  At table creation, table type json + field type record -> field type validation fails
+      ###  At streaming inserts, schema cannot be specified
+      # config_param :field_record,  :string, defualt: nil
+      # config_param :optional_data_field, :string, default: nil
+
+      REGEXP_MAX_NUM = 10
+      config_param :replace_record_key, :bool, default: false
+      (1..REGEXP_MAX_NUM).each {|i| config_param :"replace_record_key_regexp#{i}", :string, default: nil }
+
+      config_param :convert_hash_to_json, :bool, default: false
+
+      config_param :time_format, :string, default: nil
+      config_param :time_field, :string, default: nil
```

Comment:
Use `inject` plugin helper to inject timestamp string as a field to be inserted/loaded, instead of defining `time_format`, `time_field` and creating `TimeFormatter` by myself.
It handles tag and time, with timezone consideration, by providing standard parameter name (e.g., `tag_key`, `time_key`, `time_type`, `time_format`, `timezone`, `localtime`, `utc`, ...)